spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/src/main/java/org/springframework/cloud/function/adapter/gcloud/FunctionInvoker.java

```diff
@@ -55,29 +63,33 @@ public FunctionInvoker(Class<?> configurationClass) {
 	}
 
 	private void init() {
+		if (System.getenv().containsKey("spring.cloud.function.definition")) {
+			this.functionName = System.getenv("spring.cloud.function.definition");
+		}
 		System.setProperty("spring.http.converters.preferred-json-mapper", "gson");
 		Thread.currentThread() // TODO: remove after upgrading to 1.0.0-alpha-2-rc5
 				.setContextClassLoader(FunctionInvoker.class.getClassLoader());
 		initialize(null);
 	}
 
+	private <I> Function<Message<I>, Message<byte[]>> lookupFunction() {
+		Function<Message<I>, Message<byte[]>> function = this.catalog.lookup(functionName,
+				MimeTypeUtils.APPLICATION_JSON.toString());
+		Assert.notNull(function, "'function' with name '" + functionName + "' must not be null");
+		return function;
+	}
+
 	/**
-	 * The implementation of a GCF {@link HttpFunction} that will be used as the entry point from GCF.
+	 * The implementation of a GCF {@link HttpFunction} that will be used as the entry
+	 * point from GCF.
 	 */
 	@Override
 	public void service(HttpRequest httpRequest, HttpResponse httpResponse) throws Exception {
 		try {
-			String functionName = System.getenv().containsKey("spring.cloud.function.definition")
-					? System.getenv("spring.cloud.function.definition") : "";
-
-			Function<Message<BufferedReader>, Message<byte[]>> function =
-					this.catalog.lookup(functionName, MimeTypeUtils.APPLICATION_JSON.toString());
-			Assert.notNull(function, "'function' with name '" + functionName + "' must not be null");
+			Function<Message<BufferedReader>, Message<byte[]>> function = lookupFunction();
```

Comment:
do we need to lookup function every time?

Reply:
It is safe, since the first time the function is looked up it is cached, so there is really no overhead.